qiskit/transpiler/instruction_durations.py

```diff
@@ -75,26 +78,54 @@ def from_backend(cls, backend: Backend):
-            for gate, insts in backend_properties._gates.items():
-                for qubits, props in insts.items():
-                    if "gate_length" in props:
-                        gate_length = props["gate_length"][0]  # Throw away datetime at index 1
-                        instruction_durations.append((gate, qubits, gate_length, "s"))
-            for q, props in backend.properties()._qubits.items():
-                if "readout_length" in props:
-                    readout_length = props["readout_length"][0]  # Throw away datetime at index 1
-                    instruction_durations.append(("measure", [q], readout_length, "s"))
-
-        try:
-            dt = backend.configuration().dt
-        except AttributeError:
-            dt = None
-
-        return InstructionDurations(instruction_durations, dt=dt)
+        dt = None
+
+        # Logic to handle if backend is sub-class of old BackendV1
+        if isinstance(backend, BackendV1):
+            backend_properties = backend.properties()
+            if hasattr(backend_properties, "_gates"):
+                for gate, insts in backend_properties._gates.items():
+                    for qubits, props in insts.items():
+                        if "gate_length" in props:
+                            gate_length = props["gate_length"][0]  # Ignore datetime at index 1
+                            instruction_durations.append((gate, qubits, gate_length, "s"))
+                for q, props in backend.properties()._qubits.items():
+                    if "readout_length" in props:
+                        readout_length = props["readout_length"][0]  # Ignore datetime at index 1
+                        instruction_durations.append(("measure", [q], readout_length, "s"))
+
+            try:
+                dt = backend.configuration().dt
+            except AttributeError:
+                pass
+
+        # Logic to handle if backend is sub-class is BackendV2
+        elif isinstance(backend, BackendV2):
+            target = backend.target
+            inst_with_no_props = {"delay"}
+            inst_duration = None
+            for name in target.operation_names:
+                for qubits, inst_props in target._gate_map[name].items():
+                    if name in inst_with_no_props:
+                        # Setting the duration for 'delay' to zero.
+                        inst_duration = 0.0
+                    else:
+                        try:
+                            inst_duration = inst_props.duration
+                        except AttributeError:
+                            logger.info("%s on %s did not report any duration", name, qubits)
+                            continue
+                    instruction_durations.append((name, qubits, inst_duration, "s"))
+
+            try:
+                dt = target.dt
+            except AttributeError:
+                logger.info("Backend Target didn't report any dt")
```

Comment:
Why not just use `backend.target.durations()` (https://docs.quantum.ibm.com/api/qiskit/qiskit.transpiler.Target#durations)?

Reply:
@mtreinish , Haha! :) I was working on a different PR, I discovered `InstructionDurations.from_backend()` didn't have the `BackendV2` and did this PR. 
A much better way: `backend.target.durations()` didn't cross my mind :)  
I see now, this is a silly PR, so should I close this?

Reply:
@mtreinish , added suggestion, please see if this is good enough :)